transformer_engine/common/CMakeLists.txt

```
@@ -14,6 +14,12 @@ set(CMAKE_CUDA_STANDARD_REQUIRED ON)
 
 project(transformer_engine LANGUAGES CUDA CXX)
 
+if(NOT DEFINED NVTE_MAX_BUILD_THREADS)
+  set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} --threads 4")
+else()
```

I see this matches the assumption in the PyTorch and Paddle builds that the system can handle 4x more threads than `nproc`:
https://github.com/NVIDIA/TransformerEngine/blob/fa4b866d5f7159338d3367c15cc34a2d7ff96d2c/build_tools/pytorch.py#L71
https://github.com/NVIDIA/TransformerEngine/blob/fa4b866d5f7159338d3367c15cc34a2d7ff96d2c/build_tools/paddle.py#L65

However, this seems dubious given the number of users getting stuck in the build process (https://github.com/NVIDIA/TransformerEngine/issues/976, https://github.com/NVIDIA/TransformerEngine/issues/978, https://github.com/NVIDIA/TransformerEngine/issues/1077). It seems better to make sure the build works by default, and to make the fast build opt-in:
```suggestion
if(DEFINED NVTE_MAX_BUILD_THREADS)
```
It may be worthwhile updating the PyTorch and Paddle builds so they have similar logic.

---

I see. Then let's bring it into a single place only.